lib/elastic_record/as_document.rb

```diff
@@ -10,42 +10,36 @@ def as_search_document(mapping_properties = elastic_index.mapping[:properties])
       end
     end
 
-    def as_partial_update_document
-      mapping_properties = elastic_index.mapping[:properties]
+    def as_partial_update_document(mapping_properties = elastic_index.mapping[:properties])
       changed_fields = respond_to?(:saved_changes) ? saved_changes.keys : changed
```

Comment:
So there's another part to this bug where "technically" the changes on `SubWidget` aren't saved (if backed by another DB). So the bug persists. We can fix this a couple different ways, using/merging `provided` might work. However, that would change the semantics/behavior of this code. Currently noodling on this.